### PR TITLE
[Translation] Add completion feature on translation pull and push commands

### PR DESCRIPTION
**Pull request by welcoMattic**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no 
| Tickets       | Fix (partially) https://github.com/symfony/symfony/issues/43594 
| License       | MIT
| Doc PR        | 

This PR adds completion for arguments and options on `translation:pull` and `translation:push` commands.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 1]
```diff
@@ -9,15 +9,15 @@
  * file that was distributed with this source code.
  */
 
-namespace Symfony\Bundle\FrameworkBundle\Tests\Command\AboutCommand;
+namespace Symfony\Component\Translation\Tests\Command\AboutCommand;
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: This change is wrong. This file is in FrameworkBundle

Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: same in most other modified files

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php (hunk 1 above)
Comment (welcoMattic) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: Thanks, IDK where this change comes from :grimacing:

**Comment by welcoMattic** [comment]
@GromNaN It's ready for review here

File: src/Symfony/Component/Translation/Command/TranslationPushCommand.php [hunk 2]
```diff
@@ -32,27 +34,51 @@ final class TranslationPushCommand extends Command
     protected static $defaultName = 'translation:push';
     protected static $defaultDescription = 'Push translations to a given provider.';
 
-    private $providers;
+    private $providerCollection;
```
Comment (fabpot) [inline src/Symfony/Component/Translation/Command/TranslationPushCommand.php]: Let's not rename this var as the new name is not "better" than the old one.

File: src/Symfony/Component/Translation/Test/TranslationProviderTestCase.php [hunk 3]
```diff
@@ -9,7 +9,7 @@
  * file that was distributed with this source code.
  */
 
-namespace Symfony\Component\Translation\Tests\Command;
+namespace Symfony\Component\Translation\Test;
```
Comment (fabpot) [inline src/Symfony/Component/Translation/Test/TranslationProviderTestCase.php]: Moving to the Test namespace means that we allow developers to use it in their own app, with our BC policy. Not sure this is what we want here.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Translation/Command/TranslationPushCommand.php (hunk 2 above)
Comment (welcoMattic) [reply src/Symfony/Component/Translation/Command/TranslationPushCommand.php]: I did it to match with the same var in TranslationPullCommand https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Translation/Command/TranslationPullCommand.php#L36

File: src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php [hunk 4]
```diff
@@ -13,11 +13,11 @@
 
 use Symfony\Bundle\FrameworkBundle\Command\AboutCommand;
 use Symfony\Bundle\FrameworkBundle\Console\Application;
-use Symfony\Bundle\FrameworkBundle\Tests\Command\AboutCommand\Fixture\TestAppKernel;
 use Symfony\Bundle\FrameworkBundle\Tests\TestCase;
 use Symfony\Component\Console\Tester\CommandTester;
 use Symfony\Component\Filesystem\Exception\IOException;
 use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Translation\Tests\Command\AboutCommand\Fixture\TestAppKernel;
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: I don't get this change. AFAICS, this file is in the `Symfony\Bundle\FrameworkBundle\Tests\Command\AboutCommand` namespace.

Comment (welcoMattic) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Command/AboutCommand/AboutCommandTest.php]: Sure, it's a bad manipulation. I revert it.

File: src/Symfony/Component/Translation/Command/TranslationPullCommand.php [hunk 5]
```diff
@@ -52,6 +54,36 @@ public function __construct(TranslationProviderCollection $providerCollection, T
...
+        }
+
+        if ($input->mustSuggestOptionValuesFor('locales')) {
+            $suggestions->suggestValues($this->enabledLocales);
+
+            return;
+        }
+
+        if ($input->mustSuggestOptionValuesFor('format')) {
+            $suggestions->suggestValues(['php', 'xlf', 'xlf12', 'xlf20', 'po', 'mo', 'yml', 'yaml', 'ts', 'csv', 'json', 'ini', 'res']);
```
Comment (welcoMattic) [inline src/Symfony/Component/Translation/Command/TranslationPullCommand.php:83]: Shouldn't we centralize this list of translation file formats somewhere?

Comment (fabpot) [reply src/Symfony/Component/Translation/Command/TranslationPullCommand.php:83]: Not sure it's worth it. Let's keep it that way for now.

File: src/Symfony/Component/Translation/Command/TranslationPushCommand.php (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Component/Translation/Command/TranslationPushCommand.php]: Let's revert (we are always trying to minimize the diffs).

Comment (welcoMattic) [reply src/Symfony/Component/Translation/Command/TranslationPushCommand.php]: Ok, I've reverted it

**Comment by fabpot** [comment]
Can you skip the tests for older versions of Symfony? Like in #43731 for instance.

**Comment by welcoMattic** [comment]
~The last failing test seems to be related to https://github.com/symfony/symfony/pull/43688~
Rebased on 5.4, everything is green.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @welcoMattic.

**Pull request merged**